android/java/org/chromium/chrome/browser/brave_news/CardBuilderFeedCard.java

```diff
@@ -1335,28 +1340,30 @@ private void setDisplayAdImage(ImageView imageView, Image adDataImage, int index
         }
 
         final Url adImageUrl = imageUrlTemp;
-
-        mBraveNewsController.getImageData(adImageUrl, imageData -> {
-            if (imageData != null) {
-                Bitmap decodedByte = BitmapFactory.decodeByteArray(imageData, 0, imageData.length);
-                Glide.with(mActivity)
-                        .asBitmap()
-                        .load(decodedByte)
-                        .fitCenter()
-                        .priority(Priority.IMMEDIATE)
-                        .diskCacheStrategy(DiskCacheStrategy.ALL)
-                        .into(new CustomTarget<Bitmap>() {
-                            @Override
-                            public void onResourceReady(@NonNull Bitmap resource,
-                                    @Nullable Transition<? super Bitmap> transition) {
-                                imageView.setImageBitmap(resource);
-                            }
-                            @Override
-                            public void onLoadCleared(@Nullable Drawable placeholder) {}
-                        });
-                imageView.setClipToOutline(true);
-            }
-        });
+        if (mBraveNewsController != null) {
+            mBraveNewsController.getImageData(adImageUrl, imageData -> {
+                if (imageData != null) {
+                    Bitmap decodedByte =
+                            BitmapFactory.decodeByteArray(imageData, 0, imageData.length);
+                    Glide.with(mActivity)
+                            .asBitmap()
+                            .load(decodedByte)
+                            .fitCenter()
+                            .priority(Priority.IMMEDIATE)
+                            .diskCacheStrategy(DiskCacheStrategy.ALL)
```

Comment:
we shouldn't be using glide in general, but we definitely should not be allowing it to cache to disk. Also what thread does this code run on?